AMP_CACHE_GUIDELINES.md

```diff
@@ -0,0 +1,83 @@
+# AMP Cache Guidelines
+
+## Purpose
+
+Provide guidelines to external parties for implementing their own AMP Caches.
+
+## Guidelines: Core
+
+1. An AMP Cache only serves [valid AMP input documents](https://github.com/ampproject/amphtml/blob/master/spec/amp-html-format.md).
+
+2. It participates in the AMP Project's [validator release cycle](https://github.com/ampproject/amphtml/tree/master/validator).
+
+3. It participates in the AMP JS library release cycle and makes every effort to serve the latest version. It does not allow sites to perform version locking.
+
+4. Pledges to maintain URL space forever (even beyond the lifetime of the cache itself): 
+
+    4.1. This can be achieved by donating the URL space to a trustworthy third party entity such as [archive.org](http://archive.org).
+
+    4.2. This means that, should a cache decide to no longer operate, URLs should redirect to the origin URL or be served by another cache.
+
+5. Uses HTTPS serving only.
+
+6. Serves a strong [Content Security Policy](https://w3c.github.io/webappsec-csp/) ([CSP](https://developer.chrome.com/extensions/contentSecurityPolicy)) - a CSP defines a mechanism by which web developers can control the resources which a particular page can fetch or execute, as well as a number of security-relevant policy decisions.
+
+    6.1. Implements separately documented security rewrites on served content.
+
+    6.2. The CSP should be equivalent or stronger to what is being served on [cdn.ampproject.org](http://cdn.ampproject.org).
+
+    6.3. Caches must update their CSP in a timely fashion (within 7 days) at the request of the AMP Project.
+
+7. Supports a public Update ping mechanism which provides a mechanism for document publishers to notify the AMP cache about new, updated or deleted documents: 
+
+    7.1. Equivalent to the [Google AMP Cache Update ping API](https://developers.google.com/amp/cache/update-ping#update-ping-format)
+
+8. Supports a public AMP Cache URL API:
+
+    8.1. Equivalent to the [Google AMP Cache URL API](https://developers.google.com/amp/cache/overview#amp-cache-url-format)
+
+9. Provides a faithful visual and UX reproduction of source document, although visually imperceptible changes in image resolution or compression rate are acceptable.  
+
+10. No obstruction of the contents of the original AMP document. 
+
+11. No changes to ads, access, analytics, or similar.
+
+12. Respects all resource deletions within a reasonable timeframe.
```

Comment:
Is this deletion via update-ping or removing content from origin with 404/410?  I don't believe the Google Cache will ever remove content with out explicit update-ping and then 404/410?